lib/logstash/util/defaults.rb

```diff
@@ -0,0 +1,16 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/util"
+
+module LogStash module Util module Defaults
+
+  def self.filter_workers_count
+    cpu_cores
+  end
+
+  # specifics
+
+  def self.cpu_cores
+    Java::Java.lang.Runtime.getRuntime.availableProcessors
+  end
+end end end
```

Comment:
Because of the refactoring of the pipeline we will be using the `concurrent-ruby` gem a lot more and it already provide a method to return the number of cpu.

https://github.com/ruby-concurrency/concurrent-ruby/blob/master/lib/concurrent/utility/processor_counter.rb

It uses the same code as your behind the scene, they also provide a `physical_core` method.
I think we can use it directly?